solr/core/src/java/org/apache/solr/core/HttpSolrClientProvider.java

```diff
@@ -40,107 +34,55 @@
-        KNOWN_METRIC_NAME_STRATEGIES.get(UpdateShardHandlerConfig.DEFAULT_METRICNAMESTRATEGY);
-    if (cfg != null) {
-      metricNameStrategy = KNOWN_METRIC_NAME_STRATEGIES.get(cfg.getMetricNameStrategy());
-      if (metricNameStrategy == null) {
-        throw new SolrException(
-            SolrException.ErrorCode.SERVER_ERROR,
-            "Unknown metricNameStrategy: "
-                + cfg.getMetricNameStrategy()
-                + " found. Must be one of: "
-                + KNOWN_METRIC_NAME_STRATEGIES.keySet());
-      }
-    }
-    return metricNameStrategy;
+    httpSolrClient = httpClientBuilder.build();
   }
 
   private InstrumentedHttpListenerFactory.NameStrategy getNameStrategy(
       UpdateShardHandlerConfig cfg) {
-    InstrumentedHttpListenerFactory.NameStrategy nameStrategy =
-        InstrumentedHttpListenerFactory.KNOWN_METRIC_NAME_STRATEGIES.get(
-            UpdateShardHandlerConfig.DEFAULT_METRICNAMESTRATEGY);
-
-    if (cfg != null) {
-      nameStrategy =
-          InstrumentedHttpListenerFactory.KNOWN_METRIC_NAME_STRATEGIES.get(
-              cfg.getMetricNameStrategy());
-      if (nameStrategy == null) {
-        throw new SolrException(
-            SolrException.ErrorCode.SERVER_ERROR,
-            "Unknown metricNameStrategy: "
-                + cfg.getMetricNameStrategy()
-                + " found. Must be one of: "
-                + KNOWN_METRIC_NAME_STRATEGIES.keySet());
-      }
-    }
-    return nameStrategy;
+    String metricNameStrategy =
+        cfg != null && cfg.getMetricNameStrategy() != null
+            ? cfg.getMetricNameStrategy()
+            : UpdateShardHandlerConfig.DEFAULT_METRICNAMESTRATEGY;
+    return InstrumentedHttpListenerFactory.getNameStrategy(metricNameStrategy);
   }
 
-  public Http2SolrClient getSolrClient() {
+  Http2SolrClient getSolrClient() {
     return httpSolrClient;
   }
 
-  public void setSecurityBuilder(HttpClientBuilderPlugin builder) {
+  void setSecurityBuilder(HttpClientBuilderPlugin builder) {
     builder.setup(httpSolrClient);
   }
 
-  @Override
-  public void initializeMetrics(SolrMetricsContext parentContext, String scope) {
-    solrMetricsContext = parentContext.getChildContext(this);
-    String expandedScope = SolrMetricManager.mkName(scope, SolrInfoBean.Category.HTTP.name());
+  void initializeMetrics(SolrMetricsContext parentContext) {
+    var solrMetricsContext = parentContext.getChildContext(this);
+    String expandedScope =
+        SolrMetricManager.mkName(METRIC_SCOPE_NAME, SolrInfoBean.Category.HTTP.name());
     trackHttpSolrMetrics.initializeMetrics(solrMetricsContext, expandedScope);
   }
 
-  @Override
-  public SolrMetricsContext getSolrMetricsContext() {
-    return solrMetricsContext;
-  }
-
-  @Override
-  public void close() {
+  void close() {
     try {
-      SolrMetricProducer.super.close();
+      trackHttpSolrMetrics.close();
     } catch (Exception e) {
       log.error("Error closing SolrMetricProducer", e);
     }
```

Comment:
we have IOUtils.closeQuietly

Reply:
`IOUtils.closeQuietly` takes Closeable, and InstrumentedHttpListenerFactory is a sub-class of AutoCloseable. Do we have any other utility that takes AutoCloseable?

Reply:
closeQuiety could be modified to widen the parameter definition to AutoCloseable.